### PR TITLE
Migrate `AddCalendarActivity` to Jetpack Compose

### LINKED ISSUE #39: Migrate `AddCalendarActivity` to Jetpack Compose
Depends on #34

### PR DESCRIPTION
**Pull request by ArnyminerZ**
Sorry for the big PR, but splitting it into two added even more code, since a library has to be used to import the Fragment into Compose. I hope it's not a big issue 😅

### CODE REVIEW

**Review by sunkup — CHANGES_REQUESTED** [review]
> Sorry for the big PR, but splitting it into two added even more code, since a library has to be used to import the Fragment into Compose. I hope it's not a big issue 😅

No, its actually very doable. Most of the changes are very straightforward :)


But it does not seem to be working? I am getting lots of exceptions and I can't add any subscriptions ...

```
12:49:32.495  E  Couldn't create calendar
                 java.lang.NullPointerException
                 	at at.bitfire.icsdroid.model.SubscriptionModel$create$1.invokeSuspend(SubscriptionModel.kt:38)
                 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
                 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
                 	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:115)
                 	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:103)
                 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
                 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)
                 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)
                 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)
12:49:32.500  D  Compat change id reported: 147798919; UID 10194; state: ENABLED
--------- beginning of crash
12:49:32.500  D  Shutting down VM
12:49:32.505  E  FATAL EXCEPTION: main
                 Process: at.bitfire.icsdroid, PID: 4915
                 java.lang.IllegalStateException: You must either set a text or a view
                 	at com.android.internal.util.Preconditions.checkState(Preconditions.java:215)
                 	at android.widget.Toast.show(Toast.java:199)
                 	at at.bitfire.icsdroid.ui.AddCalendarActivity$onCreate$3.invoke(AddCalendarActivity.kt:118)
                 	at at.bitfire.icsdroid.ui.AddCalendarActivity$onCreate$3.invoke(AddCalendarActivity.kt:117)
                 	at at.bitfire.icsdroid.ui.AddCalendarActivity$sam$androidx_lifecycle_Observer$0.onChanged(Unknown Source:2)
                 	at androidx.lifecycle.LiveData.considerNotify(LiveData.java:133)
                 	at androidx.lifecycle.LiveData.dispatchingValue(LiveData.java:151)
                 	at androidx.lifecycle.LiveData.setValue(LiveData.java:309)
                 	at androidx.lifecycle.MutableLiveData.setValue(MutableLiveData.java:50)
                 	at androidx.lifecycle.LiveData$1.run(LiveData.java:93)
                 	at android.os.Handler.handleCallback(Handler.java:958)
                 	at android.os.Handler.dispatchMessage(Handler.java:99)
                 	at android.os.Looper.loopOnce(Looper.java:205)
                 	at android.os.Looper.loop(Looper.java:294)
                 	at android.app.ActivityThread.main(ActivityThread.java:8177)
                 	at java.lang.reflect.Method.invoke(Native Method)
                 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
                 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
```

Edit: Now I could add a subscription, but the subscription name is the exception? :thinking: 

image

File: app/src/main/java/at/bitfire/icsdroid/ui/AddCalendarActivity.kt [hunk 1]
```diff
@@ -4,33 +4,95 @@
...
 import at.bitfire.icsdroid.model.SubscriptionSettingsModel
+import at.bitfire.icsdroid.model.ValidationModel
+import com.google.accompanist.themeadapter.material.MdcTheme
+import kotlinx.coroutines.launch
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import java.net.URI
+import java.net.URISyntaxException
 
-class AddCalendarActivity: AppCompatActivity() {
+@OptIn(ExperimentalFoundationApi::class)
```
Comment (sunkup) [inline app/src/main/java/at/bitfire/icsdroid/ui/AddCalendarActivity.kt:59]: What is this doing? Do we need it?

> This foundation API is experimental and is likely to change or be removed in the future

Sounds dangerous :sweat_smile:

**Comment by ArnyminerZ** [comment]
🤔 I'll take a look, but it was working fine for me. I did try it only with API 31 though, I will try now with API 34.

**Comment by ArnyminerZ** [comment]
It's working for me on emulator API 31 and 34, in two different computers, and on my testing device (Xiaomi Mi A2 API 29) 🤔 

Can you try building again? Clearing caches or something? Also the errors are pretty strange

**Comment by sunkup** [comment]
> It's working for me on emulator API 31 and 34, in two different computers, and on my testing device (Xiaomi Mi A2 API 29) 🤔
> 
> Can you try building again? Clearing caches or something? Also the errors are pretty strange

Ah, yes. Not sure why the quirks happened ... Will review now, but prob. not ready till Monday :)

File: app/src/main/java/at/bitfire/icsdroid/ui/AddCalendarActivity.kt (hunk 1 above)
Comment (sunkup) [reply app/src/main/java/at/bitfire/icsdroid/ui/AddCalendarActivity.kt:59]: Ah, its for this one
https://developer.android.com/jetpack/compose/layouts/pager

Hmm, I guess we really have to use it ... Would have thought compose to be more mature ...

**Review by sunkup — APPROVED** [review]
Looks good :)

**Pull request merged**